pam/internal/adapter/gdmmodel.go

```diff
@@ -271,3 +289,12 @@ func (m gdmModel) changeStage(s proto.Stage) tea.Cmd {
 		return nil
 	}
 }
+
+func (m gdmModel) stopConversations() gdmModel {
+	// We're about to exit: let's ensure that all the messages have been processed.
+	time.Sleep(gdmPollFrequency * 2)
+	for gdm.ConversationInProgress() {
```

Comment:
isn’t that a busy loop? If so, I think we should have a timeout with channel.

Reply:
Yeah, I was thinking about it... It's also true that if we fail here we should probably panic, no?

Reply:
We should never panic is a library, always return an error. Don’t panic in Go, in general

Reply:
Well, as programmer error... Even though this is not the case, so I'll warn only.